src/html/angular-i18n-translations-extractor.ts

```diff
@@ -0,0 +1,83 @@
+import {AngularElement, HtmlTranslationExtractionContext} from "./html-translation-extractor";
+import {Attribute} from "./element-context";
+
+const I18N_ATTRIBUTE_REGEX = /^i18n-.*$/;
+const I18N_ATTRIBUTE_NAME = "i18n";
+const ID_INDICATOR = "@@";
+
+export default function angularI18nTranslationsExtractor(
+    element: AngularElement,
+    context: HtmlTranslationExtractionContext
+): void {
+    const i18nElementTranslation = element.attributes.find(attribute => attribute.name === I18N_ATTRIBUTE_NAME);
+
+    if (i18nElementTranslation) {
+        handleTranslationsOfElements(element, context, i18nElementTranslation);
+    }
+
+    const i18nAttributeTranslation = element.attributes.filter(attribute => I18N_ATTRIBUTE_REGEX.test(attribute.name));
+
+    handleTranslationsOfAttributes(element, context, i18nAttributeTranslation);
+}
+
+function handleTranslationsOfElements(
+    element: AngularElement,
+    context: HtmlTranslationExtractionContext,
+    attribute: Attribute
+): void {
+    const translationId = extractTranslationId(attribute, context);
+
+    let defaultTranslation: string;
+    if (element.texts.length > 0 && translationId) {
+        defaultTranslation = element.texts[0].text;
+        context.registerTranslation({
+            translationId: translationId,
+            defaultText: defaultTranslation,
+            position: element.startPosition
+        });
+    } else {
+        context.emitError(`The element ${context.asHtml()} with attribute  ${attribute.name} is empty and is therefore missing the default translation.`, attribute.startPosition);
+    }
+}
+
+function handleTranslationsOfAttributes(
+    element: AngularElement,
+    context: HtmlTranslationExtractionContext,
+    i18nAttributes: Attribute[]
+): void {
+
+    for (const i18nAttribute of i18nAttributes) {
+        const translationId = extractTranslationId(i18nAttribute, context);
```

Comment:
I suggest extracting the for block content into a separated function. This would allow a quicker return when the  are no `translationId`, `attributeName` or `defaultText`. In this way the some of the checks can be removed.